### PR TITLE
[SPARK-45342][BUILD] Remove the scala doc compilation option specific to Scala 2.12

### PR DESCRIPTION
**Pull request by LuciferYang**

### CODE REVIEW

File: project/SparkBuild.scala [hunk 1]
```diff
@@ -358,7 +358,7 @@ object SparkBuild extends PomBuild {
       "-doc-title", "Spark " + version.value.replaceAll("-SNAPSHOT", "") + " ScalaDoc"
     ) ++ {
       // Do not attempt to scaladoc javadoc comments under 2.12 since it can't handle inner classes
-      if (scalaBinaryVersion.value == "2.12") Seq("-no-java-comments") else Seq.empty
+      if (scalaBinaryVersion.value == "2.13") Seq("-no-java-comments") else Seq.empty
```
Comment (LuciferYang) [inline project/SparkBuild.scala:361]: @srowen @easel I've found a leftover Scala 2.12 version check here, but when using the check method provided in https://github.com/apache/spark/pull/20042: `sbt tags/publishLocal`, I couldn't distinguish the scala doc built with or without `-no-java-comments`. I hope to get your help,  how should we determine whether to keep the option `-no-java-comments` for Scala 2.13? Thanks ~

Comment (srowen) [reply project/SparkBuild.scala:361]: I think you should just remove this part, right? does not apply in Scala 2.13

Comment (LuciferYang) [reply project/SparkBuild.scala]: Thanks for your help @srowen , but I found that there are errors when executing `build/sbt doc -Phadoop-3 -Phadoop-cloud -Pmesos -Pyarn -Pkinesis-asl -Phive-thriftserver -Pspark-ganglia-lgpl -Pkubernetes -Phive -Pvolcano`. Let me fix them first.

**Comment by LuciferYang** [comment]
we can use https://github.com/apache/spark/pull/43158, close this one

**Pull request closed without merging**